docarray/array/mixins/find.py

```diff
@@ -96,7 +96,7 @@ def find(
         limit: Optional[Union[int, float]] = 20,
         metric_name: Optional[str] = None,
         exclude_self: bool = False,
-        filter: Optional[Dict] = None,
+        filter: Optional[Union[Dict, str]] = None,
```

Comment:
```suggestion
        filter: Union[Dict, str, None] = None,
```

Not sure if this is not a bit better like this

Reply:
I have always preferred the Optional

Reply:
I prefer it as well for normal use but for Optional[Union[]] I am not sure